### ISSUE TITLE
cBranch and cLocalBranch are empty, cHash has wrong value

### ISSUE THREAD

**Issue by knolleCC** [gitextensions/gitextensions#10223]
### Environment

- Git Extensions 3.5.4.12724
- Build 65f01f39982ccb121c4df7d4b00b506939553958
- Git 2.37.3.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET Framework 4.8.4515.0

### Issue description

I use the `cBranch` variable in my GitExt Scripts and it worked fine in version 3.4.3.
After upgrading to 3.5.4 the variable is always empty :(
I tryed `cLocalBranch` then, but it is also empty.
I then tryed `cHash` but it strangely returns a hash of some other commit.

### Steps to reproduce

Create a Script that echos cBranch, cLocalBranch or cHash

### Did this work in previous version of GitExtensions?

3.4.3.9999

**Comment by mstv** [comment]
TL;DR: Perhaps you want to use `{sBranch}`, `{sLocalBranch}` and `{sHash}`.

`{cOption}` yields info about the `c`urrent checkout.
`{sOption}` yields info about the `s`elected commit / revision.

I think that's why the hash is unexpected for you.
And I guess, you tried this with a detached HEAD checkout, i.e. there is no `c`urrent branch.

So most likely, there remains only one minor issue: `{cBranch}` and `{cLocalBranch}` seem to yield the *last* checked out branch. The last value is kept when checking out a commit as detached HEAD. The value should be invalidated or rather be replaced with the checked out hash.

**Comment by knolleCC** [comment]
Thanks for your answer @mstv but no, I was not on a detached head and no, I cannot use sBranch as a replacement, because I actually want the current branch, which will not be the selected commit in my script, so sBranch will not help. 
As I sad, I used the script quite a long while with GitExt 3.4.3 and it worked just as I expected. The problem seems to be the new GitExt or the updated Git itself. I also updated Git when I updated GitExt, so maybe I will try and check with an older Git again. 

But please tell me, what does cOption return exactly?

**Comment by mstv** [comment]
I used "Option" just as a placeholder for "Branch", "LocalBranch", "Hash", etc.

Yes, you are right. The info is not updated anymore - after first use.

**Comment by mstv** [comment]
If #10225 is no issue for you, you could try the bugfix based on v4 beta https://ci.appveyor.com/project/gitextensions/gitextensions/builds/44982488/artifacts.

**Comment by knolleCC** [comment]
Ok, thank you! Will there also be another bugfix release for 3.x containing this fix?

**Comment by RussKie** [comment]
No, 3.x is done.
